Azure.ResourceManager.Core/Resources/ResourceIdentifier.cs

```diff
@@ -143,6 +141,10 @@ public override string ToString()
         /// <param name="id">A properly formed resource identity</param>
         protected virtual void Parse(string id)
         {
+            if (id == null)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
             // Throw for null, empty, and string without the correct form
             if (string.IsNullOrWhiteSpace(id) || !id.Contains('/'))
                 throw new ArgumentOutOfRangeException($"'{id}' is not a valid resource");
```

Comment:
Should be ArgumentException.

Reply:
Bring up at meeting in general to go over parsing ID